### PR TITLE
[Gutenberg] Emit editor save events

### LINKED ISSUE #5666: VideoPress block: Limit synchronization of block attributes to when post has just been saved
The web version of the block only syncs changes to the block's settings when a post has been manually saved ([ref](https://github.com/Automattic/jetpack/blob/125163a2575cec14371c30294dad657313cf9e1a/projects/packages/videopress/src/client/block-editor/hooks/use-sync-media/index.ts#L279)). However, there isn't currently a built-in way to check when a post has just been saved on iOS or Android. 

To address this, I'm imagining that we would:

* Set up a listener that receives a message over the bridge when a post is saved on the native side.
* Either directly listen for the aforementioned message in [the `useSyncMedia` hook](https://github.com/Automattic/jetpack/blob/6b6c4f5fa932723198fba4be769c62aa0a4b0480/projects/packages/videopress/src/client/block-editor/hooks/use-sync-media/index.ts) or, alternatively, connect this to a native version of [the `isSavingPost` selector](https://github.com/WordPress/gutenberg/blob/2929720235e4f8537424b0a4fcbde4a06eda23a2/packages/editor/src/store/selectors.js#L677-L692).

When we have implemented a way to detect when a post has just been saved on native, we can then limit the VideoPress block's synchronization accordingly.

### PR DESCRIPTION
**Pull request by SiobhyB**
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/5666

## Related PRs

* `Gutenberg`: https://github.com/WordPress/gutenberg/pull/49781
* `Gutenberg Mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5663
* `Jetpack`: https://github.com/Automattic/jetpack/pull/30131
* `Android`: https://github.com/wordpress-mobile/WordPress-Android/pull/18279

## To Test

After making some edits to a post or page, there are multiple flows to manually save based on whether a post/page is a draft, published, or scheduled. These have been outlined below.

<details>
 <summary>Flows for saving a draft ⤵️</summary>

* Tap the **Update** button.
* Tap the X icon to exit the post and **Update Draft**.
* Tap the three dots to the upper right and then **Publish**.
* The the three dots to the upper right to get to Post Settings. From here, set the post's publish date and then **Schedule**.

</details>

<details>
 <summary>Flows for saving a published post ⤵️</summary>

* Tap the **Update** button.
* Tap the X icon to exit the post and **Update Post**.
* The the three dots to the upper right to get to Post Settings. From here, set the post's publish date and then **Schedule**.

</details>

<details>
 <summary>Flows for saving a scheduled post ⤵️</summary>

* Tap the **Update** button.
* Tap the X icon to exit the post and **Update Post**.
* The the three dots to the upper right to get to Post Settings. From here, set the post's publish date and then **Schedule**.

</details>

## Regression Notes
1. Potential unintended areas of impact

As this PR is only emitting a save event, and not changing any functionality, I can't foresee any unintended impact.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Although I don't believe there will be unintended impact, I did some quick manual smoke testing of the editor's functionality and saving as part of this PR.

3. What automated tests I added (or what prevented me from doing so)

N/A

<hr />

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by fluiddot — COMMENTED** [review]
@SiobhyB I can't build the app locally to test these changes, I'm getting the same failure as [in the CI jobs](https://buildkite.com/automattic/wordpress-ios/builds/13761#01879157-565d-47d7-9726-3f09b7be5b86/6-5489) when building the project. Would you mind taking a look?

**Comment by SiobhyB** [comment]
@fluiddot, whoops, sorry about that, that should be fixed as of https://github.com/wordpress-mobile/WordPress-iOS/pull/20522/commits/d7584cea3350e88e1898b5b0e8f25d1462f4c95b (there was a second function call, which I overlooked as it had been removed in my local set up 🤦‍♀️).

File: WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift [hunk 1]
```diff
@@ -57,6 +59,10 @@ extension PublishingEditor {
 
     }
 
+    func emitPostSaveEvent() {
+
+    }
+
```
Comment (fluiddot) [inline WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift]: I tried to remove this function to check if it's needed, and got compilation errors in `AztecPostViewController` and `StoryEditor`. I think it's ok to define the default behavior here as it's optional, but I'm wondering if we could add a comment about why this function is empty, WDYT?

File: WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift [hunk 2]
```diff
@@ -95,6 +101,8 @@ extension PublishingEditor {
 
         mapUIContentToPostAndSave(immediate: true)
 
+        emitPostSaveEvent()
```
Comment (fluiddot) [inline WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift]: I noticed that some lines below the publishing can be canceled, so maybe we should call this function once we are sure that the post will be saved, WDYT?

https://github.com/wordpress-mobile/WordPress-iOS/blob/d7584cea3350e88e1898b5b0e8f25d1462f4c95b/WordPress/Classes/ViewRelated/Post/PostEditor%2BPublish.swift#L106-L110

I potential place would be right before we track the `PostSave` event:
https://github.com/wordpress-mobile/WordPress-iOS/blob/d7584cea3350e88e1898b5b0e8f25d1462f4c95b/WordPress/Classes/ViewRelated/Post/PostEditor%2BPublish.swift#L151-L167

File: WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift (hunk 1 above)
Comment (SiobhyB) [reply WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift]: Sure thing! I followed the same pattern that I saw for other optional functions like this ([example](https://github.com/wordpress-mobile/WordPress-iOS/blob/d7584cea3350e88e1898b5b0e8f25d1462f4c95b/WordPress/Classes/ViewRelated/Post/PostEditor%2BPublish.swift#L58-L60)), which don't have comments, but agree it's a good idea to add for clarification.

Comment (SiobhyB) [reply WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift]: Added in https://github.com/wordpress-mobile/WordPress-iOS/pull/20522/commits/07ddf84736968bc5bd9f782088186f68323a499a, I took the liberty of also adding the same comment to the other empty function in the file, for consistency.

File: WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift (hunk 2 above)
Comment (SiobhyB) [reply WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift]: That logic makes sense by me! Updated in https://github.com/wordpress-mobile/WordPress-iOS/pull/20522/commits/4e944ebdd90613e7a0960d37f64f5c4afe5bbb92.

File: Podfile.lock [hunk 3]
```diff
@@ -4,21 +4,21 @@ PODS:
...
-  - AppCenter/Core (4.4.1)
-  - AppCenter/Crashes (4.4.1):
+  - AppCenter/Core (4.4.3)
+  - AppCenter/Crashes (4.4.3):
     - AppCenter/Core
-  - AppCenter/Distribute (4.4.1):
+  - AppCenter/Distribute (4.4.3):
     - AppCenter/Core
   - Automattic-Tracks-iOS (0.13.0):
     - Sentry (~> 7.25)
```
Comment (fluiddot) [inline Podfile.lock]: @SiobhyB I noticed that several pods have been updated. However, I don't think this is expected when only bumping the Gutenberg Mobile reference. Usually, only the pods related to Gutenberg are the ones that are updated 🤔.

Comment (fluiddot) [reply Podfile.lock]: Not sure if it's caused by this, but I'm getting the following error when building the app:
```
./WordPress-iOS/WordPress/Classes/Utility/Kanvas/KanvasCameraAnalyticsHandler.swift:5:20: error build: Type 'KanvasAnalyticsHandler' does not conform to protocol 'KanvasAnalyticsProvider'
```

Comment (SiobhyB) [reply Podfile.lock]: Hmmm, I'm not sure how those got updated along the way 🤔 this should be addressed in the most recent commits, though.

Comment (fluiddot) [reply Podfile.lock]: Great, I've just checked the last changes and it looks good. Thanks!

**Review by fluiddot — APPROVED** [review]
Approved via https://github.com/Automattic/jetpack/pull/30131#pullrequestreview-1392487317.

**NOTE:** I've only tested these changes building the app locally.

File: Podfile [hunk 4]
```diff
@@ -91,7 +91,7 @@ def shared_style_pods
 end
 
 def gutenberg_pods
-  gutenberg tag: 'v1.93.0'
+  gutenberg commit: 'a9bafbb3e6ecf1b0fd2532af5a72a5756f5d1690'
```
Comment (fluiddot) [inline Podfile]: Heads up that the Gutenberg Mobile reference doesn't point to the last commit from https://github.com/wordpress-mobile/gutenberg-mobile/pull/5663: [`3b7427e1f12a6e34f5b5760c29a3db91cca33857`](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5663/commits/3b7427e1f12a6e34f5b5760c29a3db91cca33857)

**Pull request merged**